lib/hydra/derivatives/tempfile_service.rb

```diff
@@ -0,0 +1,47 @@
+require 'mime/types'
+
+module Hydra::Derivatives
+  class TempfileService
+
+    def self.create(file, &block)
+      new(file).tempfile(&block)
+    end
+
+    attr_reader :source_file
+
+    def initialize(source_file)
+      @source_file = source_file
+    end
+
+    def tempfile(&block)
+      if source_file.respond_to? :to_tempfile
+        source_file.send(:to_tempfile, &block)
+      elsif source_file.has_content?
+        default_tempfile(&block)
+      end
+    end
+
+    def default_tempfile(&block)
+      Tempfile.open(filename_for_characterization) do |f|
+        f.binmode
+        if source_file.content.respond_to? :read
+          f.write(source_file.content.read)
+        else
+          f.write(source_file.content)
+        end
+        source_file.content.rewind if source_file.content.respond_to? :rewind
+        f.rewind
+        yield(f)
+      end
+    end
+
+    def filename_for_characterization
```

Comment:
that's just what the name was before I moved the method to this class. Got a better suggestion?